hpx/parallel/util/loop.hpp

```diff
@@ -96,23 +261,88 @@ namespace hpx { namespace parallel { namespace util
                 return it;
             }
         };
+
+
+        template <typename T>
+        struct loop_n <prefetching_iterator<T>>
+        {
+            ///////////////////////////////////////////////////////////////////
+            // handle sequences of non-futures when using prefetching
+            template <typename F>
+            static prefetching_iterator<T> call(prefetching_iterator<T> it, std::size_t count, F && f)
+            {
+                std::size_t cnt = count + 1;
+                for (/**/; cnt != 0; (void) --cnt, ++it)
+                {
+                    std::vector<std::size_t>::iterator inner_it = it.base;
+                    std::size_t j = it.idx;
+
+                    for(/**/; j < std::min( (it.range_size ),(it.idx + it.chunk_size) ); ++j)
+                    {
+                        f(inner_it);
+                        ++inner_it;
+                    }
+
+                    if(j < it.range_size - 1  )
+                        for (auto& x: it.M_)
+                            _mm_prefetch(((char*)(&x[j+1])), _MM_HINT_T0);
```

Comment:
This needs to get handled with a preprocessor conditional. It is a SSE intrinsic and not available on all patforms